lib/presentation/mixins/play_video_action_mixin.dart

```diff
@@ -1,20 +1,41 @@
+import 'package:fluffychat/utils/interactive_viewer_gallery.dart';
 import 'package:fluffychat/utils/platform_infos.dart';
+import 'package:fluffychat/widgets/hero_dialog_route.dart';
 import 'package:fluffychat/widgets/video_viewer_desktop_theme.dart';
 import 'package:fluffychat/widgets/video_viewer_mobile_theme.dart';
 import 'package:flutter/material.dart';
 
 mixin PlayVideoActionMixin {
-  void playVideoAction(BuildContext context, String uriOrFilePath) async {
-    await showDialog(
-      context: context,
-      useRootNavigator: PlatformInfos.isWeb,
-      useSafeArea: false,
-      builder: (context) {
-        if (PlatformInfos.isWeb || PlatformInfos.isDesktop) {
-          return VideoViewerDesktopTheme(path: uriOrFilePath);
-        }
-        return VideoViewerMobileTheme(path: uriOrFilePath);
-      },
-    );
+  void playVideoAction(
+    BuildContext context,
+    String uriOrFilePath, {
+    String? eventId,
+  }) async {
+    if (!PlatformInfos.isWeb) {
+      Navigator.of(context).push(
```

Comment:
Can you explain why do you use it?

Reply:
Hero transitions are only enabled for transitions between two instances of PageRoute. Reference: https://stackoverflow.com/questions/44403417/hero-animation-with-an-alertdialog. So I need to create an extends class `HeroPageRoute` then push into Navigator.